homework-5/homework-5.py

```diff
@@ -0,0 +1,86 @@
+# Backend Programming with Python
+# Homework 5 - Battleships
+
+import random, math
+
+
+class SpaceShip:
+    def __init__(self, name, hp, armour, damage):
+        self.name = name
+        self.hp = hp
+        self.armour = armour
+        self.damage = damage
+
+    def attack(self, other):
+        if isinstance(other, SpaceShip):
+            damage_dealt = self.damage - other.armour
+            if damage_dealt < 0 < self.damage:
```

Comment:
Why are you doing a chained comparison here? No need to check for self.damage here. It is enough to do:
```if damage_dealt < 0```